assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js

```diff
@@ -87,6 +118,46 @@ const DashboardIdeasWidget = ( {
 	);
 	const activeTab = DashboardIdeasWidget.tabIDsByIndex[ activeTabIndex ];
 
+	const [ ideaHubContainer, inView ] = useInView( {
+		triggerOnce: true,
+		threshold: 0.25,
+	} );
+
+	useEffect( () => {
+		if ( inView ) {
+			trackEvent( IDEA_HUB_GA_CATEGORY_WIDGET, 'widget_view' );
+		}
+	}, [ inView ] );
+
+	useEffect( () => {
+		if ( trackedWidgetView ) {
+			return;
+		}
+
+		if ( hasNoIdeas ) {
+			trackEvent(
+				IDEA_HUB_GA_CATEGORY_WIDGET,
+				'widget_gathering_data_view'
+			);
+		} else if ( hasManyIdeas ) {
+			trackEvent(
+				IDEA_HUB_GA_CATEGORY_WIDGET,
+				'default_tab_view',
+				DashboardIdeasWidget.tabIDsByIndex[ activeTabIndex ]
+			);
+		}
+
+		if ( hasNoIdeas || hasManyIdeas ) {
+			setTrackedWidgetView( true );
+		}
+	}, [
+		hasNoIdeas,
+		hasManyIdeas,
+		trackedWidgetView,
+		setTrackedWidgetView,
+		activeTabIndex,
+	] );
```

Comment:
This implementation is slightly different from what's described in the ACs (and what we do in the PSI widget): One of these two events should be sent right after `widget_view`. Right now they're sent regardless of whether the user scrolls to the widget or not.

Reply:
Updated.